### PR TITLE
Modify DefaultCredentialsProvider to use reflection to access AE SDK

### PR DESCRIPTION
**Pull request by mziccard**
This PR does the following:
- Remove `OAuth2Utils.exceptionWithCause`, use exception costructor instead
- Add package-private `AppEngineCredentials` to `google-auth-library-oauth2-http` that uses reflection to access AE SDK
- Modify `DefaultCredentialsProvider` to instantiate the new `AppEngineCredentials` class

/cc @garrettjonesgoogle @anthmgoogle

### CODE REVIEW

**Comment by mziccard** [comment]
Guys any comment on this?

**Comment by anthmgoogle** [comment]
@garrettjonesgoogle and I should both review this. I will review it today.

File: oauth2_http/java/com/google/auth/oauth2/OAuth2Utils.java [hunk 1]
```diff
@@ -94,17 +94,6 @@ static boolean headersContainValue(HttpHeaders headers, String headerName, Strin
   }
 
   /**
-   * Throw the exception with the specified cause.
-   *
-   * <p>Needed instead of constructor version to be compatible with JDK 1.5 which is required
-   * until Andriod libraries can update to JDK 1.6
-   */
-  static <T extends Throwable> T exceptionWithCause(T exception, Throwable cause) {
```
Comment (anthmgoogle) [inline oauth2_http/java/com/google/auth/oauth2/OAuth2Utils.java]: Removing this seems like good cleanup. It was needed before because of Android constraints which seem to be more relaxed now. It seems tangential to this change. Perhaps factor this into a separate PR?

Comment (mziccard) [inline oauth2_http/java/com/google/auth/oauth2/OAuth2Utils.java]: I opened #72 for this, PTAL.

**Comment by anthmgoogle** [comment]
That other change looks good. I'm still LGTM on this one.

**Comment by mziccard** [comment]
@anthmgoogle I rebased to catch latest changes. @garrettjonesgoogle do you also want to have a look?

**Comment by garrettjonesgoogle** [comment]
LGTM
